app/src/main/java/chat/rocket/android/layouthelper/ExtRealmModelListAdapter.java

```diff
@@ -37,6 +41,35 @@ protected void notifyRealmModelItemChanged(int position) {
     notifyItemChanged(position + 1);
   }
 
+  public void setLayoutManager(RecyclerView.LayoutManager layoutManager) {
+    this.layoutManager = layoutManager;
+  }
+
+  protected ListUpdateCallback listUpdateCallback = new ListUpdateCallback() {
+    @Override
+    public void onInserted(int position, int count) {
+      notifyItemRangeInserted(position + 1, count);
+      if (layoutManager != null) {
+        layoutManager.scrollToPosition(0);
+      }
```

Comment:
My concern is that it is better to implement with "observer" here.

![slice](https://cloud.githubusercontent.com/assets/11763113/21934685/aaa25a80-d9ed-11e6-9783-b7560fa86ade.png)

For example this kind of improvement is hard to implement with extending RecyclerView adapter.